frontend/src/metabase/visualizations/echarts/visualizations/PieChart/mixins.ts

```diff
@@ -0,0 +1,103 @@
+import type { SunburstSeriesOption } from "echarts";
+
+import type { EChartsMixin } from "metabase/visualizations/types";
+import {
+  computeLabelDecimals,
+  formatPercent,
+} from "metabase/visualizations/visualizations/PieChart/utils";
+import { formatValue } from "metabase/lib/formatting";
+import { getMetricIndex, getSlices } from "./utils";
+
+export const pieSeriesMixin: EChartsMixin = ({ option, props }) => {
+  const slices = getSlices({ props });
```

Comment:
I'm starting doubting the mixins approach in favor of having a function per chart for building the options object. With the mixins approach we cannot share intermediate computations which does not look critical on Pie chart but on a line/area/bar chart with 2k data points per 10 series it can affect performance significantly. Let's chat about it tomorrow